### ISSUE TITLE
用自己训练的densenet模型检测，识别不出汉字？

### ISSUE THREAD

**Issue by JJoohhnnn** [YCG09/chinese_ocr#209]
150万张数据集图片，acc :98%， loss:0.05  可是检测的时候什么字都识别不出来是为什么啊？

**Comment by JJoohhnnn** [comment]
没有keras.h5预训练模型

**Comment by boris-lb** [comment]
我的也是自己数据，只有数字，train和val精度都在0.98以上，但是测试精度为0，预测的值都是一个或者两个的数字，麻烦问一下你解决了吗？

**Comment by sail75** [comment]
我也遇到这个问题，也是自己的数据，train和val 精度都很高，但是预测的时候几乎都发生了错误。我不知道为什么会发生这样的情况，请问您解决这个问题了吗？如果解决了可以告知我是什么原因促使这个错误发生的吗？

**Comment by JJoohhnnn** [comment]
> 我的也是自己数据，只有数字，train和val精度都在0.98以上，但是测试精度为0，预测的值都是一个或者两个的数字，麻烦问一下你解决了吗？

你是模型加载错了吗

**Comment by JJoohhnnn** [comment]
> 我也遇到这个问题，也是自己的数据，train和val 精度都很高，但是预测的时候几乎都发生了错误。我不知道为什么会发生这样的情况，请问您解决这个问题了吗？如果解决了可以告知我是什么原因促使这个错误发生的吗？

你的数据集是语料库里的连续一句话还是随机的汉字？

**Comment by boris-lb** [comment]
> > 我的也是自己数据，只有数字，train和val精度都在0.98以上，但是测试精度为0，预测的值都是一个或者两个的数字，麻烦问一下你解决了吗？
> 
> 你是模型加载错了吗

模型没有加载错，就是我训练好的模型。

**Comment by sail75** [comment]
> > 我也遇到这个问题，也是自己的数据，train和val 精度都很高，但是预测的时候几乎都发生了错误。我不知道为什么会发生这样的情况，请问您解决这个问题了吗？如果解决了可以告知我是什么原因促使这个错误发生的吗？
> 
> 你的数据集是语料库里的连续一句话还是随机的汉字？

您好，谢谢您的关心！目前我已经找到问题了。我的训练代码是自己写的，我对数据进行了归一化处理。然后编写测试代码的时候，归一化的地方写错了。。。。。尴尬😅，进行修改之后结果正确。

**Comment by Maruidear** [comment]
你好 我的是自己数据，只有数字，train和val精度都很高，预测的时候感觉用的依旧是原来的模型，麻烦问一下这是什么问题？

**Comment by Maruidear** [comment]
> > > 我也遇到这个问题，也是自己的数据，train和val 精度都很高，但是预测的时候几乎都发生了错误。我不知道为什么会发生这样的情况，请问您解决这个问题了吗？如果解决了可以告知我是什么原因促使这个错误发生的吗？
> > 
> > 
> > 你的数据集是语料库里的连续一句话还是随机的汉字？
> 
> 您好，谢谢您的关心！目前我已经找到问题了。我的训练代码是自己写的，我对数据进行了归一化处理。然后编写测试代码的时候，归一化的地方写错了。。。。。尴尬，进行修改之后结果正确。



> > > 我也遇到这个问题，也是自己的数据，train和val 精度都很高，但是预测的时候几乎都发生了错误。我不知道为什么会发生这样的情况，请问您解决这个问题了吗？如果解决了可以告知我是什么原因促使这个错误发生的吗？
> > 
> > 
> > 你的数据集是语料库里的连续一句话还是随机的汉字？
> 
> 您好，谢谢您的关心！目前我已经找到问题了。我的训练代码是自己写的，我对数据进行了归一化处理。然后编写测试代码的时候，归一化的地方写错了。。。。。尴尬，进行修改之后结果正确。

请问你训练densenet网络的时候有修改代码吗？我用自己的数据训练网络，结果运行demo.py依旧用的原模型

**Comment by ericosmic** [comment]
@sail75   请问你是使用原有的预训练权重，还是重新训练呢？

**Comment by ericosmic** [comment]
@Maruidear   我也只用数字来进行fine-tune训练，但是得到的模型只能识别数字了。 你的问题解决了吗

**Comment by boris-lb** [comment]
我的问题解决了，是图片的输入高度32太小了，我根据实际情况修改输入图片高度64以上，问题解决。

**Comment by JJoohhnnn** [comment]
> 我的问题解决了，是图片的输入高度32太小了，我根据实际情况修改输入图片高度64以上，问题解决。

为什么图片高度太小就不行呢?

**Comment by Maruidear** [comment]
> @Maruidear 我也只用数字来进行fine-tune训练，但是得到的模型只能识别数字了。 你的问题解决了吗

我的问题没有解决，我训练出的模型数字都识别不出来，全是空格。请问你的数据量是多少？

**Comment by boris-lb** [comment]
> > 我的问题解决了，是图片的输入高度32太小了，我根据实际情况修改输入图片高度64以上，问题解决。
> 
> 为什么图片高度太小就不行呢?

具体原因我也不清楚，就算图片卷积到最后feature map太小，也解释不通为什么训练集和验证集精度比较高，而测试集精度几乎为0，更何况densenet是共用前面的feature。

**Comment by sail75** [comment]
@Maruidear 我只使用网络结构哪一个文件，因为要适应我的输入，所以训练代码和测试代码我都重新写了。

**Comment by sail75** [comment]
@ericosmic 我重新训练的权重。

**Comment by foundersix** [comment]
请问大佬们们训练时，图片的数据集时咋制作的？

**Comment by PeterBishop0** [comment]
> 你好 我的是自己数据，只有数字，train和val精度都很高，预测的时候感觉用的依旧是原来的模型，麻烦问一下这是什么问题？

请问你怎么处理标签的，想请教一下

**Comment by ql106** [comment]
> > > 我也遇到这个问题，也是自己的数据，train和val 精度都很高，但是预测的时候几乎都发生了错误。我不知道为什么会发生这样的情况，请问您解决这个问题了吗？如果解决了可以告知我是什么原因促使这个错误发生的吗？
> > 
> > 
> > 你的数据集是语料库里的连续一句话还是随机的汉字？
> 
> 您好，谢谢您的关心！目前我已经找到问题了。我的训练代码是自己写的，我对数据进行了归一化处理。然后编写测试代码的时候，归一化的地方写错了。。。。。尴尬😅，进行修改之后结果正确。

请问你是如何归一化的

**Comment by 666888985** [comment]
@foundersix ，@PeterBishop0，@ql106 ，@foundersix  各位大佬那个densenet+ctc部分的测试代码有吗？我看项目里面只给了ctpn的检测测试代码没有给densenet+ctc识别部分的测试代码，各位大佬可否发我一份测试代码，谢谢

**Comment by yoummiegao** [comment]
> > @Maruidear 我也只用数字来进行fine-tune训练，但是得到的模型只能识别数字了。 你的问题解决了吗
> 
> 我的问题没有解决，我训练出的模型数字都识别不出来，全是空格。请问你的数据量是多少？

@Maruidear请问您后来这个问题，是如何解决的？我这边RGB finetune 精度也还可以，但是也是识别结果就是一堆空白。
